UnitsNet.Serialization.JsonNet.Tests/UnitsNetIQuantityJsonConverterTest.cs

```diff
@@ -0,0 +1,125 @@
+        public void
+            UnitsNetIQuantityJsonConverter_WriteJson_throws_ArgumentNullException_when_arguments_are_null(JsonWriter writer, JsonSerializer serializer, string parameterName)
+        {
+            var exception = Assert.Throws<ArgumentNullException>(() => sut.WriteJson(writer, Power.FromWatts(10D), serializer));
+
+            Assert.Equal($"Value cannot be null.\r\nParameter name: {parameterName}", exception.Message);
+        }
+
+        [Fact]
+        public void UnitsNetIQuantityJsonConverter_WriteJson_works_with_NULL_value()
+        {
+            var result = new StringBuilder();
+            using(var stringWriter = new StringWriter(result))
+            using (var writer = new JsonTextWriter(stringWriter))
+            {
+                sut.WriteJson(writer, null, JsonSerializer.CreateDefault());
+            }
+
+            Assert.Equal("null", result.ToString());
+        }
+
+        [Fact]
+        public void UnitsNetIQuantityJsonConverter_WriteJson_works_as_expected()
+        {
+            var result = new StringBuilder();
+
+            using (var stringWriter = new StringWriter(result))
+                using(var writer = new JsonTextWriter(stringWriter))
+            {
+                sut.WriteJson(writer, Power.FromWatts(10.2365D), JsonSerializer.CreateDefault());
+            }
+
+            Assert.Equal("{\"Unit\":\"PowerUnit.Watt\",\"Value\":10.2365}", result.ToString());
+        }
+
+        [Fact]
+        public void UnitsNetIQuantityJsonConverter_CanRead_returns_true()
+        {
+            Assert.True(sut.CanRead);
+        }
+
+        public static IEnumerable<object[]> ReadJson_NullArguments => new[]
+        {
+            new object[] { null, JsonSerializer.CreateDefault(), "reader" },
+            new object[] { new JsonTextReader(new StringReader("")), null, "serializer" }
+        };
+
+        [Theory]
+        [MemberData(nameof(ReadJson_NullArguments))]
+        public void UnitsNetIQuantityJsonConverter_ReadJson_throws_ArgumentNullException_when_arguments_are_null(JsonReader reader, JsonSerializer serializer, string paramName)
+        {
+            var exception = Assert.Throws<ArgumentNullException>(() => sut.ReadJson(reader, typeof(IQuantity), null, false, serializer));
+
+            Assert.Equal($"Value cannot be null.\r\nParameter name: {paramName}", exception.Message);
+        }
+
+        [Fact]
+        public void UnitsNetIQuantityJsonConverter_ReadJson_handles_NULL_values_correctly()
+        {
+            var json = "null";
+
+            using(var stringReader = new StringReader(json))
+            using (var jsonReader = new JsonTextReader(stringReader))
+            {
+                var result = sut.ReadJson(jsonReader, typeof(IQuantity), null, false, JsonSerializer.CreateDefault());
+
+                Assert.Null(result);
+            }
+        }
+
+        [Fact]
+        public void UnitsNetIQuantityJsonConverter_ReadJson_works_as_expected()
```

Comment:
Should we add a test case about not supporting deserialization of `IComparable` anymore?

Reply:
we are supporting it, but in a different converter?
The obsolete message on the old converter should reflect this